### DISCUSSION TITLE
Subscriptions implementation

### DISCUSSION THREAD

**Discussion by igoraugustynski** [swellstores/community#296] [Help] [answered]
Hi,
I need to implement the possibility to sell digital subscriptions in a store using Angular and swell-js. 
How should those work? Can I add a subscription to a cart?
Currently I'm using `swell.cart.addItem()` with `purchaseOption: { type: 'subscription' }` but it seems to work as if it was a one-time purchase, not a subscription. Also, it ignores the trial days. 
The swell-js documentation mentions a `swell.subscriptions.create()` should I use this, without a cart, and write a different checkout process for subscriptions? If so, how do I go about setting billing information (address, cards, etc) to this subscription?
In general, the documentation is lacking in this regard, or I couldn't find relevant information.

**Comment by sags95** [comment]
Hey @igoraugustynski, can you share how you've configured the plans for the product in the dashboard? You should be able to add an item with a purchase option to the cart just by specifying the type.

**Reply by igoraugustynski** [reply]
<img width="1268" alt="251203180-8b0d34bf-4ac1-4fe1-86d3-b9eb121da0a8" src="https://github.com/swellstores/community/assets/94485687/7e2b4b0a-3410-4bd1-ae2e-1b0a64c2aff4">
Here is a screenshot of the relevant configuration. Just to clarify: The product is added to the cart, but it seems it's treated like a product with standard pricing: The subTotal goes up to 10$ (which it shouldn't, because of the trial days) and after placing the order I don't see anything under "subscriptions" in the dashboard.

**Reply by logeshswell** [reply, marked as answer]
@igoraugustynski I tried the following code to add the same product into the cart and it is working fine for me.
```ts
const itemDetail = {
  productId: '64a31b10dd248d00116a693d',
  purchase_option: {
    type: 'subscription'
  }
}
await swell.cart.addItem(itemDetail)
  .then(response => {
    console.log(response);
  }).catch(async error => {
    console.log(error);
  });
```

[New cart](https://parsifal-commerce.swell.store/admin/carts/64a7dfe1c076d5001226b6d4) is created as follows and it is the expected behavior.

<img width="880" alt="Screenshot 2023-07-07 at 3 25 33 PM" src="https://github.com/swellstores/community/assets/100766131/d59df022-1187-43d1-a338-cda59c473170">


If you have more than 1 plan for the subscription product, then you need to pass `plan_id` as shown below:

<img width="799" alt="Screenshot 2023-07-07 at 3 29 03 PM" src="https://github.com/swellstores/community/assets/100766131/986c1bc2-8f26-48e9-b9e5-1d2bae2efd8b">

https://developers.swell.is/frontend-api/carts#add-an-item-with-a-purchase-option

**Reply by logeshswell** [reply]
If you continue to experience this issue, please share the swell-js version and sample code that you use to add an item to the cart.
